solr/solrj/src/java/org/apache/solr/client/solrj/cloud/autoscaling/Clause.java

```diff
@@ -117,10 +117,10 @@ private Clause(Map<String, Object> m) {
     strict = Boolean.parseBoolean(String.valueOf(m.getOrDefault("strict", "true")));
     Optional<String> globalTagName = m.keySet().stream().filter(Policy.GLOBAL_ONLY_TAGS::contains).findFirst();
     if (globalTagName.isPresent()) {
-      globalTag = parse(globalTagName.get(), m);
-      if (m.size() > 2) {
-        throw new RuntimeException("Only one extra tag supported for the tag " + globalTagName.get() + " in " + toJSONString(m));
+      if (m.size() > 3) {
```

Comment:
from my very limited understanding of this code it appears that something like
```
    if (globalTagName.isPresent()) {
      validateGlobalTag(m, globalTagName.get());
      globalTag = parse(globalTagName.get(), m);
 ...
and 
...
  private void validateGlobalTag(Map<String, Object> m, String tagName) {
    if (m.size() > 2) {
      if (!m.containsKey("strict") || m.size() > 3) {
        throw new RuntimeException("Only, 'strict' and one extra tag supported for the tag " + tagName + " in " + toJSONString(m));
      }
    }
  }
```
would be the real intention here... but it would be great if @sigram could confirm that.

Reply:
@gus-asf yes, that's the intent, though your pseudo-code is still incorrect - if there's no `strict` tag then `m.size() > 2` is already an error. In other words, for global tags we expect exactly 2 keys (tag and operand), with optional third key `strict`.

Reply:
I think that's the logic Gus used. It's equivalent to:

```java
private void validateGlobalTag(Map<String, Object> m, String tagName) {
    if (m.size() > 2) {
      if (!(m.containsKey("strict") && m.size() == 3)) {
        throw new RuntimeException("Only, 'strict' and one extra tag supported for the tag " + tagName + " in " + toJSONString(m));
      }
    }
  }
```

This will error unless there are exactly 3 keys and once of them is `strict`.